pkg/telemetry/phonehome/telemeter.go

```diff
@@ -8,16 +8,27 @@ type Telemeter interface {
 	// the buffers.
 	Stop()
 	// Identify updates the user traits.
-	Identify(userID string, props map[string]any)
+	Identify(props map[string]any)
 	// Track registers an event, caused by the user.
-	Track(event, userID string, props map[string]any)
+	Track(event string, props map[string]any)
 	// Group adds the user to a group, supplying group specific properties.
-	Group(groupID, userID string, props map[string]any)
+	Group(groupID string, props map[string]any)
+
+	// IdentifyUserAs updates the user traits.
+	IdentifyUserAs(userID, clientID, clientType string, props map[string]any)
+	// TrackUserAs registers an event, caused by the user.
+	TrackUserAs(userID, clientID, clientType, event string, props map[string]any)
+	// GroupUserAs adds the user to a group, supplying group specific properties.
+	GroupUserAs(userID, clientID, clientType, groupID string, props map[string]any)
 }
 
 type nilTelemeter struct{}
 
-func (t *nilTelemeter) Stop()                                              {}
-func (t *nilTelemeter) Identify(userID string, props map[string]any)       {}
-func (t *nilTelemeter) Track(event, userID string, props map[string]any)   {}
-func (t *nilTelemeter) Group(groupID, userID string, props map[string]any) {}
+func (t *nilTelemeter) Stop()                            {}
+func (t *nilTelemeter) Identify(_ map[string]any)        {}
+func (t *nilTelemeter) Track(_ string, _ map[string]any) {}
+func (t *nilTelemeter) Group(_ string, _ map[string]any) {}
+
+func (t *nilTelemeter) IdentifyUserAs(_, _, _ string, _ map[string]any) {}
+func (t *nilTelemeter) TrackUserAs(_, _, _, _ string, _ map[string]any) {}
```

Comment:
Shouldn't this be called `TrackAsUser`? Otherwise it sounds like we are tracking the user.

Reply:
The second two parameters are client impersonation, so the full name could have been `TrackUserEventAsClient`.
Anyway, what does tracking a user mean? Watching their steps, and that's essentially what this method does: it tracks what a user is doing. If it makes sense, I'd keep the name.